src/main/java/org/opensearch/performanceanalyzer/collectors/telemetry/RTFHeapMetricsCollector.java

```diff
@@ -54,52 +55,57 @@ public void collectMetrics(long startTime) {
             return;
         }
 
-        initialiseMetrics();
+        initialiseMetricsIfNeeded();
```

Comment:
Can we initialise in the constructor itself?

Reply:
We can't. Metrics registry object is not set by the time these collectors are initialized. Collectors are initialized as the part of PerformanceAnalyzerPlugin constructor.

Reply:
Ok.